### ISSUE TITLE
No write access in .gotron directory

### ISSUE THREAD

**Issue by nanohard** [Equanox/gotron#23]
I got a `permission denied` error at the start of creating files in the directory `.gotron`.
Turns out the `.gotron` directory, and every subdirectory, is created with permissions of `555` instead of `755`.

EDIT
My `umask` is `0022`.

```
⟩ go run main.go
panic: : : : mkdir .gotron/assets: permission denied: mkdir .gotron/assets: permission denied: : mkdir .gotron/assets: permission denied: mkdir .gotron/assets: permission denied: : : mkdir .gotron/assets: permission denied: mkdir .gotron/assets: permission denied: : mkdir .gotron/assets: permission denied: mkdir .gotron/assets: permission denied

goroutine 1 [running]:
main.main()
        /code/go/src/github.com/nanohard/NoX/main.go:24 +0x109
exit status 2

⟩ la
total 12K
drwxr-xr-x 1 user users   60 Feb 11 23:15 ./
drwxr-xr-x 1 user users  110 Feb 11 22:19 ../
drwxr-xr-x 1 user users   98 Feb 11 23:15 .git/
-rw-r--r-- 1 user users  279 Feb 11 22:45 go.mod
-rw-r--r-- 1 user users 2.9K Feb 11 22:45 go.sum
dr-xr-xr-x 1 user users    0 Feb 11 23:15 .gotron/
-rw-r--r-- 1 user users  757 Feb 11 22:23 main.go
```

**Comment by Equanox** [comment]
Thx for your report. #24 will create the directory explicitly with correct permission.

**Comment by nanohard** [comment]
That fixed the top-level directory `.gotron`, but the subdirectories still have the permission issue.
Also, the 2 files it creates are `444` but should be `644`. Not sure why this is happening.
Is it just me?

**Comment by Equanox** [comment]
No it's not just you, but it's not 100% reproducable. We'll try to provide another fix in the next couple of hours.

**Comment by puengel** [comment]
Hi @nanohard ,
Trying to reproduce your problem. With yesterdays fix #24 .gotron and all subdirectories have `755` permissions. Also I did not find a single file with `444`. Can you please specify which files and directories are in bad state for you? Try something like `find -perm 444` and `find -perm 555` in `.gotron` and paste the result here. This should help us trace where the problem is coming from since files are being copied from different locations.

**Comment by nanohard** [comment]
```
- .gotron (755)
  -- assets (555)
    --- index.html (444)
    --- index.js (444)
```

By manually running `chmod` on `assets`, running again, then doing the same for the files everything installs properly.

**Comment by puengel** [comment]
Are you running `gotron.New()` or `gotron.New("path-to-your-ui-folder")`?
First one will create some `index.html` and `index.js` for you (these should have `644` already).
Second one takes and copies files from `path-to-your-ui-folder` into assets folder with same permissions as in previous folder. So if your `index.html` is in `444` in your source folder it will be in `444` in the destination folder as well.

Can you reproduce the issue when you delete the `.gotron` folder and run `go run main.go`?

**Comment by nanohard** [comment]
I'm running `gotron.New()`, the code from the README.
I created a new project folder just to make sure as well.
I am using modules as well.
```
go mod init
go get github.com/Equanox/gotron
(put example code into main.go)
go run main.go
```
The end result is I end up with:
```
- .gotron (755)
  -- assets (555)
```
Then I chmod `assets` to `755`.
Run `go run main.go` again.
Then chmod `assets/index.html` and `assets/index.js` to `644`.
Run `go run main.go` again.
Profit!

**Comment by puengel** [comment]
Ok found the problem. `go get` does not set permissions for files as they are in github. Working on a solution to set correct permissions when copying.

**Comment by puengel** [comment]
Hi @nanohard ,
I provided a fix in #25 .
Can you check if that fixes the problem for you?
You can do that by changing 
`require github.com/Equanox/gotron v0.2.21`
to
`require github.com/Equanox/gotron v0.2.22-0.20190213153145-5cc8ae190ec0`
in your `go.mod`file and calling `go get ./...`

**Comment by nanohard** [comment]
Works without issue. Thanks!
